File: app/src/main/java/com/example/megaport/mynews/Controllers/Activities/WebViewActivity.java
package com.example.megaport.mynews.Controllers.Activities;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import com.example.megaport.mynews.R;

public class WebViewActivity extends Activity {

    @SuppressLint("SetJavaScriptEnabled")
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_webview);
        Intent intent = getIntent();
        // Retrieves the URL of the article
        String url = intent.getStringExtra("Url");
        // Finds the WebView object
        WebView webView = findViewById(R.id.webView1);
        // Assigns a client to the WebView
        webView.setWebViewClient(new AuthClient());
        webView.getSettings().setJavaScriptEnabled(true);
        // Load the URL of the article
        webView.loadUrl(url);
    }

    private class AuthClient extends WebViewClient {
        @Override
        public void onPageFinished(WebView v, String url) {
            v.clearHistory();
        }
    }
}

File: app/src/main/java/com/example/megaport/mynews/controllers/activities/WebViewActivity.kt
package com.example.megaport.mynews.controllers.activities

import android.annotation.SuppressLint
import android.app.Activity
import android.os.Bundle
import android.webkit.WebView
import android.webkit.WebViewClient
import com.example.megaport.mynews.R

class WebViewActivity : Activity() {

    @SuppressLint("SetJavaScriptEnabled")
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_webview)
        val intent = intent
        // Retrieves the URL of the article
        val url = intent.getStringExtra("Url")
        // Finds the WebView object
        val webView = findViewById<WebView>(R.id.webView1)
        // Assigns a client to the WebView
        webView.webViewClient = AuthClient()
        webView.settings.javaScriptEnabled = true
        // Load the URL of the article
        webView.loadUrl(url)
    }

    private inner class AuthClient : WebViewClient() {
        override fun onPageFinished(v: WebView, url: String) {
            v.clearHistory()
        }
    }
}

